specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2023-03-01-preview/mongoCluster.json

```diff
@@ -0,0 +1,983 @@
+          ],
+          "description": "The password of the administrator login."
+        },
+        "serverVersion": {
+          "type": "string",
+          "description": "The Mongo DB server version. Defaults to the latest available version if not specified."
+        },
+        "connectionString": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The default mongo connection string for the cluster."
+        },
+        "earliestRestoreTime": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Earliest restore timestamp in UTC ISO8601 format."
+        },
+        "provisioningState": {
+          "$ref": "#/definitions/MongoClusterProvisioningState",
+          "description": "A provisioning state of the mongo cluster.",
+          "readOnly": true
+        },
+        "clusterStatus": {
+          "$ref": "#/definitions/MongoClusterStatus",
+          "description": "A status of the mongo cluster."
+        },
+        "nodeGroupSpecs": {
+          "$ref": "#/definitions/NodeGroupSpecList",
+          "description": "The list of node group specs in the cluster."
+        }
+      }
+    },
+    "NodeGroupSpecList": {
+      "type": "array",
+      "description": "The list of node group specifications for the cluster. Must include one node group spec with kind = 'Shard'.",
+      "items": {
+        "$ref": "#/definitions/NodeGroupSpec"
+      },
+      "x-ms-identifiers": [],
+      "minItems": 1,
+      "maxItems": 1,
+      "uniqueItems": true
+    },
+    "NodeGroupSpec": {
+      "type": "object",
+      "description": "Specification for a node group.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/NodeGroupProperties"
+        }
+      ],
+      "properties": {
+        "kind": {
+          "$ref": "#/definitions/NodeKind",
+          "description": "The node type deployed in the node group."
+        },
+        "nodeCount": {
+          "type": "integer",
+          "format": "int32",
+          "description": "The number of nodes in the node group."
+        }
+      }
+    },
+    "NodeKind": {
+      "type": "string",
+      "description": "The kind of a node in the mongo cluster.",
+      "enum": [
+        "Shard"
+      ],
+      "x-ms-enum": {
+        "name": "NodeKind",
+        "modelAsString": true
+      }
+    },
+    "NodeGroupProperties": {
+      "type": "object",
+      "description": "The properties of the node group on a cluster.",
+      "properties": {
+        "sku": {
+          "type": "string",
+          "description": "The resource sku for the node group. This defines the size of CPU and memory that is provisioned for each node. Example values: 'M30', 'M40'."
+        },
+        "diskSizeGB": {
+          "type": "integer",
+          "format": "int64",
+          "description": "The disk storage size for the node group in GB. Example values: 128, 256, 512, 1024."
+        },
+        "enableHa": {
```

Comment:
>enableHa [](http://example.com/codeflow?start=9&length=8)

This bool design seems to assume you won't ever invent more ways/flavors of doing HA...

Reply:
At this stage there is no long-term plan to change the flavors. Though we plan to review this assumption during public-preview and make a call before GA.